pacstall

```diff
@@ -530,8 +531,37 @@ for i in "${@}"; do
     unset j
 done
 
-# Set the new list of arguments
-set -- "${argument_list[@]}"
+# Remove duplicates
+declare -A arg_map=(
+    ["--install"]="-I"     ["--search"]="-S"   ["--remove"]="-R"          ["--download"]="-D"
+    ["--add-repo"]="-A"    ["--update"]="-U"   ["--list"]="-L"            ["--upgrade"]="-Up"
+    ["--query-info"]="-Qi" ["--tree"]="-T"     ["--version"]="-V"         ["--help"]="-h"
+    ["--build"]="-B"       ["--keep"]="-K"     ["--disable-prompts"]="-P" ["--nocheck"]="-Nc"
+)
+declare -A hashed_arguments
+unique_argument_list=()
```

Comment:
you can (and maybe should) `unset unique_arguments_list` here instead of setting an empty array. When you do `+=($arg)`, the first one will automatically initialize it.